File: BACK_TICKETS/app/Models/Company.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Company extends Model
{
    use HasFactory;

    protected $primaryKey = 'company_id';
    protected $fillable = ['name', 'NIT', 'phone_number'];

    public function travels()
    {
        return $this->hasMany(Travel::class, 'company_id');
    }

    public function buses()
    {
        return $this->hasMany(Bus::class, 'company_id');
    }
}

File: Back_Spring/tickets/src/main/java/com/example/tickets/entity/Company.java
package com.example.tickets.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Set;
import java.util.UUID;

@NoArgsConstructor
@Getter
@Setter
@Entity
public class Company {
    @Id
    @GeneratedValue
    private UUID companyId;

    private String name;
    private String nit;
    private String phoneNumber;

//    @OneToMany(mappedBy = "company")
//    private Set<Travel> travels;
//
//    @OneToMany(mappedBy = "company")
//    private Set<Bus> buses;

    // Getters y Setters
}

